compiler/rustc_resolve/src/late.rs

```diff
@@ -940,7 +942,8 @@ impl<'a: 'ast, 'ast, 'tcx> Visitor<'ast> for LateResolutionVisitor<'a, '_, 'ast,
                         this.visit_generics(generics);
 
                         let declaration = &sig.decl;
-                        let async_node_id = sig.header.asyncness.opt_return_id();
+                        let async_node_id =
```

Comment:
var name

Reply:
I added `gen_fn_lifetime_capture.rs`, but I'm not sure it completely exercises everything it should.